pkg/leader/leader.go

```diff
@@ -109,7 +109,33 @@ func Become(ctx context.Context, lockName string) error {
 			log.Info("Became the leader.")
 			return nil
 		case apierrors.IsAlreadyExists(err):
-			log.Info("Not the leader. Waiting.")
+			existing := &corev1.ConfigMap{}
+			key := crclient.ObjectKey{Namespace: ns, Name: lockName}
+			err := client.Get(ctx, key, existing)
+			if err != nil {
+				return err
+			}
+			leaderPod := &corev1.Pod{}
+			key = crclient.ObjectKey{Namespace: ns, Name: existing.ObjectMeta.OwnerReferences[0].Name}
+			err = client.Get(ctx, key, leaderPod)
+			if err != nil {
+				return err
+			}
+
+			podFailed := leaderPod.Status.Phase == corev1.PodFailed
+			podEvicted := leaderPod.Status.Reason == "Evicted"
+			if podFailed && podEvicted {
+				log.Info("Operator pod with leader lock has been evicted.", "Leader pod", leaderPod.Name)
+				err := client.Delete(ctx, leaderPod)
+				if err == nil {
+					log.Info("Evicted leader pod has been deleted.")
+				} else {
+					log.Error(err, "Evicted leader pod could not be deleted")
```

Comment:
```suggestion
					log.Error(err, "Evicted leader pod could not be deleted: %v", err)
```

Reply:
Also if it could be just in the case of error.
if err != nil {
     //log error
}

Reply:
`err` is already captured in the log statement as the first argument. `log.Error` does not call `os.Exit` so we still need to check both conditions.